Error: Cannot find module 'node-fetch'

Hey,

I am looking into openupm as an alternative package distribution for Unity. I am getting the following error when running the openupm-cli on a Mac (Apple-Sillicon).

```
node:internal/modules/cjs/loader:1222
  throw err;
  ^

Error: Cannot find module 'node-fetch'
Require stack:
- /usr/local/lib/node_modules/openupm-cli/lib/io/check-url.js
- /usr/local/lib/node_modules/openupm-cli/lib/cli/index.js
- /usr/local/lib/node_modules/openupm-cli/lib/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1219:15)
    at Module._load (node:internal/modules/cjs/loader:1045:27)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:215:24)
    at Module.require (node:internal/modules/cjs/loader:1304:12)
    at require (node:internal/modules/helpers:123:16)
    at Object.<anonymous> (/usr/local/lib/node_modules/openupm-cli/lib/io/check-url.js:7:38)
    at Module._compile (node:internal/modules/cjs/loader:1467:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1551:10)
    at Module.load (node:internal/modules/cjs/loader:1282:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/usr/local/lib/node_modules/openupm-cli/lib/io/check-url.js',
    '/usr/local/lib/node_modules/openupm-cli/lib/cli/index.js',
    '/usr/local/lib/node_modules/openupm-cli/lib/index.js'
  ]
}

Node.js v22.4.1
```

I have tried to use openupm-cli with different versions of node:

* v18.20.4
* v20.15.1
* v22.4.1 

All to the same result. Any ideas?

I will download the source code and try to take a look myself when I have some time.

--- a/package.json
+++ b/package.json
@@ -42,7 +42,6 @@
     "@types/jest": "^29.5.12",
     "@types/libnpmsearch": "^2.0.4",
     "@types/node": "^18.19.31",
-    "@types/node-fetch": "^2.6.9",
     "@types/npmlog": "^4.1.4",
     "@types/pkginfo": "^0.4.1",
     "@types/promptly": "^3.0.3",

--- a/package.json
+++ b/package.json
@@ -72,7 +72,7 @@
     "is-wsl": "^2.2.0",
     "libnpmsearch": "^5.0.3",
     "mkdirp": "^3.0.1",
-    "node-fetch": "^3.3.2",
+    "node-fetch": "^2.7.0",
     "npm-registry-fetch": "^13.1.1",
     "npmlog": "^6.0.2",
     "pkginfo": "^0.4.1",

--- a/package.json
+++ b/package.json
@@ -73,6 +73,7 @@
     "is-wsl": "^2.2.0",
     "libnpmsearch": "^5.0.3",
     "mkdirp": "^3.0.1",
+    "node-fetch": "^3.3.2",
     "npm-registry-fetch": "^13.1.1",
     "npmlog": "^6.0.2",
     "pkginfo": "^0.4.1",

